--- keyboard_shortcuts/keyboard_shortcuts.php	2010-08-27 00:53:39.344131217 +0300
+++ keyboard_shortcuts/keyboard_shortcuts.php	2010-09-02 23:21:13.404213768 +0300
@@ -28,6 +28,7 @@
  * a:	Select all visible messages
  * A:	Mark all as read (as Google Reader)
  * c:	Compose new message
+ * d:	Delete message
  * f:	Forward message
  * j:	Go to previous page of messages (as Gmail)
  * k:	Go to next page of messages (as Gmail)
@@ -43,6 +44,7 @@
  * U:   Expand Unread
  *
  * Shortcuts, mail view:
+ * d:	Delete message
  * f:	Forward message
  * j:	Go to previous message (as Gmail)
  * k:	Go to next message (as Gmail)
@@ -84,6 +86,7 @@
         $c .= "<div class='shortcut_key'>a</div> ".$this->gettext('selectallvisiblemessages')."<br class='clear' />";
         $c .= "<div class='shortcut_key'>A</div> ".$this->gettext('markallvisiblemessagesasread')."<br class='clear' />";
         $c .= "<div class='shortcut_key'>c</div> ".$this->gettext('compose')."<br class='clear' />";
+        $c .= "<div class='shortcut_key'>d</div> ".$this->gettext('deletemessage')."<br class='clear' />";
         $c .= "<div class='shortcut_key'>f</div> ".$this->gettext('forwardmessage')."<br class='clear' />";
         $c .= "<div class='shortcut_key'>j</div> ".$this->gettext('previousmessages')."<br class='clear' />";
         $c .= "<div class='shortcut_key'>k</div> ".$this->gettext('nextmessages')."<br class='clear' />";
@@ -111,6 +114,7 @@
           $c .= "</div>";
         }
         $c .= "<div><h4>".$this->gettext("messagesdisplaying")."</h4>";
+        $c .= "<div class='shortcut_key'>d</div> ".$this->gettext('deletemessage')."<br class='clear' />";
         $c .= "<div class='shortcut_key'>f</div> ".$this->gettext('forwardmessage')."<br class='clear' />";
         $c .= "<div class='shortcut_key'>j</div> ".$this->gettext('previousmessage')."<br class='clear' />";
         $c .= "<div class='shortcut_key'>k</div> ".$this->gettext('nextmessage')."<br class='clear' />";
--- keyboard_shortcuts/keyboard_shortcuts.js	2010-08-27 00:54:10.584128202 +0300
+++ keyboard_shortcuts/keyboard_shortcuts.js	2010-08-27 09:24:53.508571945 +0300
@@ -66,6 +66,9 @@
 				case 99:		// c = compose
 					rcmail.command('compose');
 					return false;
+				case 100:		// d = delete
+					rcmail.command('delete');
+  				return false;
 				case 102:		// f = forward
 					if (rcmail.message_list.selection.length == 1)
 						rcmail.command('forward');
@@ -100,6 +103,9 @@
   			case 99:		// c = compose
   				rcmail.command('compose');
   				return false;
+				case 100:		// d = delete
+					rcmail.command('delete');
+  				return false;
 				case 102:		// f = forward
 					rcmail.command('forward');
 					return false;
